src/path_converters.h

```diff
@@ -991,8 +991,36 @@ class PathSimplifier : protected EmbeddedQueue<9>
     }
 };
 
+// XXX: remove!!!
+// #include<iostream>
+
+class SketchBase
+{
+    static int previous_seed;
+  protected:
+    /* handle the special-case value -1 which means: use the
+       previous seed plus one. If not a special value, only 
+       remember it as the last one and return it.
+    */
+    int get_prng_seed(int seed)
+    {
+        if (seed==-1) {
+            previous_seed++;
+        } else {
+            previous_seed = seed;
+        }
```

Comment:
This is there for a special case when the seed is given as -1, the seed is automatically incremented for every call as discussed in https://github.com/matplotlib/matplotlib/issues/13479. I have tried passing seed value as 0 in the tests, replacing the baseline_images and then running the tests again, but it still failed.

Reply:
Tried recreating the original image from https://github.com/matplotlib/matplotlib/issues/13479 with seed set to -1 and got this
![line_dashes](https://github.com/matplotlib/matplotlib/assets/77437286/15765025-cb75-48af-821b-c793eb126724)

Which I believe is not the desired image since all of the legends are identical to one another. Will try and find why this is the case and why the code is not doing what is is supposed to, but will probably need the help of @eudoxos since he was the one who proposed the idea.

Reply:
Hi, I tried to build from your branch and run this script below:
```py
import matplotlib.pyplot as plt
from matplotlib import rcParams
with plt.xkcd(randomness=40,scale=1,seed=-1):
    print(rcParams['path.sketch'])
    import matplotlib
    from matplotlib import gridspec
    pat,txt=plt.pie([10,20,30,40],wedgeprops={'edgecolor':'black'})
    plt.legend(pat,['first','second','third','fourth'],loc='best')
plt.show()
```
and I get something which looks correct (legend patches are different):

![image](https://github.com/matplotlib/matplotlib/assets/1029876/00034eae-d2dc-41ed-8f19-dacd63c403d1)

What does not work for me is `seed=0` where each run should produce identical results, but does not.

Reply:
Alright, I see better now: you are discaring the seed parameters https://github.com/AryanSheka/matplotlib/blob/85ec267095a42565464d6e755304eb77efe68e06/src/path_converters.h#L1038 (cf https://github.com/eudoxos/matplotlib/blob/f6404d476f2b5f2b23eb4c2ff14908d2f41a469a/src/path_converters.h#L972).

```c++
          /* if scale==0. (e.g. with test), then seed is 0
             that would set previous_seed to 0 which is not what we want.
          */
          m_seed0( scale!=0 ? get_prng_seed(seed) : 0)
```

And then, fixing that, I get this error at runtime:
```
ImportError: /home/eudoxos/build/matplotlib/lib/matplotlib/backends/_backend_agg.cpython-310-x86_64-linux-gnu.so: undefined symbol: _ZN10SketchBase13previous_seedE
```
as the `path_converters.cpp` is compiled into the `matplotlib._path` extension, but the Sketch template is now used in `_backend_agg`. So as a quick hack, I added it to `setupext.py` like this (so it is both in `_path` and `_backend_agg`:

```py
        ext = Extension(
            "matplotlib.backends._backend_agg", [
                "src/py_converters.cpp",
                "src/path_converters.cpp",
                "src/_backend_agg.cpp",
                "src/_backend_agg_wrapper.cpp",
            ])
```

and then it works: with `seed==0`, all patches are the same, with `seed==-1`, the are an always-same random sequence. I am not sure anymore if this was the idea (rather than having more randomness), but at least it does what it should.

I don't know how to integrate the static variable into the newer module separation, but I hope this track is useful for you.

Reply:
This is with `scale=3,seed=0`:
![image](https://github.com/matplotlib/matplotlib/assets/1029876/f8dc79d6-95f2-49e2-8bcb-b00580929af5)
`scale=3,seed=-1`:
![image](https://github.com/matplotlib/matplotlib/assets/1029876/87d21108-caa7-4210-8953-b5813b63d2c7)

Reply:
> ETA: reason I'm asking is that previous_seed is uninitialized at compile time and then added to every time -1 is called.
It is a static variable, initialized in https://github.com/matplotlib/matplotlib/blob/85ec267095a42565464d6e755304eb77efe68e06/src/path_converters.cpp

Reply:
> Alright, I see better now: you are discaring the seed parameters https://github.com/AryanSheka/matplotlib/blob/85ec267095a42565464d6e755304eb77efe68e06/src/path_converters.h#L1038 (cf https://github.com/eudoxos/matplotlib/blob/f6404d476f2b5f2b23eb4c2ff14908d2f41a469a/src/path_converters.h#L972).
> 
> ```c++
>           /* if scale==0. (e.g. with test), then seed is 0
>              that would set previous_seed to 0 which is not what we want.
>           */
>           m_seed0( scale!=0 ? get_prng_seed(seed) : 0)
> ```
> 
> And then, fixing that, I get this error at runtime:
> 
> ```
> ImportError: /home/eudoxos/build/matplotlib/lib/matplotlib/backends/_backend_agg.cpython-310-x86_64-linux-gnu.so: undefined symbol: _ZN10SketchBase13previous_seedE
> ```
> 
> as the `path_converters.cpp` is compiled into the `matplotlib._path` extension, but the Sketch template is now used in `_backend_agg`. So as a quick hack, I added it to `setupext.py` like this (so it is both in `_path` and `_backend_agg`:
> 
> ```python
>         ext = Extension(
>             "matplotlib.backends._backend_agg", [
>                 "src/py_converters.cpp",
>                 "src/path_converters.cpp",
>                 "src/_backend_agg.cpp",
>                 "src/_backend_agg_wrapper.cpp",
>             ])
> ```
> 
> and then it works: with `seed==0`, all patches are the same, with `seed==-1`, the are an always-same random sequence. I am not sure anymore if this was the idea (rather than having more randomness), but at least it does what it should.
> 
> I don't know how to integrate the static variable into the newer module separation, but I hope this track is useful for you.

Ah yes, I do remember discarding that line of code because of the error. Today I tried adding it back again but was unable to fix the error. Thanks a lot for clarifying it .